front-end/src/components/Reactionbar/ReactionButton.tsx

```diff
@@ -0,0 +1,144 @@
+// Copyright 2019-2020 @paritytech/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { ApolloQueryResult } from 'apollo-client';
+import React, { useContext } from 'react';
+import styled from '@xstyled/styled-components';
+
+import { UserDetailsContext } from '../../context/UserDetailsContext';
+import { NotificationContext } from '../../context/NotificationContext';
+import Button from '../../ui-components/Button';
+import { NotificationStatus } from '../../types';
+import {
+	useAddPostReactionMutation,
+	useAddCommentReactionMutation,
+	useDeletePostReactionMutation,
+	useDeleteCommentReactionMutation,
+
+	DiscussionPostAndCommentsQueryVariables,
+	ProposalPostAndCommentsQueryVariables,
+	ReferendumPostAndCommentsQueryVariables,
+	MotionPostAndCommentsQueryVariables,
+	TreasuryProposalPostAndCommentsQueryVariables,
+
+	DiscussionPostAndCommentsQuery,
+	ProposalPostAndCommentsQuery,
+	ReferendumPostAndCommentsQuery,
+	MotionPostAndCommentsQuery,
+	TreasuryProposalPostAndCommentsQuery
+} from '../../generated/graphql';
+
+export interface ReactionButtonProps {
+	className?: string
+	reactionId: number
+	reaction: string
+	count: number
+	userMap: { [ key: string ]: string; }
+	postId?: number
+	commentId?: string
+	refetch?: (variables?:
+		ReferendumPostAndCommentsQueryVariables |
+		DiscussionPostAndCommentsQueryVariables |
+		ProposalPostAndCommentsQueryVariables |
+		MotionPostAndCommentsQueryVariables |
+		TreasuryProposalPostAndCommentsQueryVariables |
+		undefined) =>
+		Promise<ApolloQueryResult<TreasuryProposalPostAndCommentsQuery>> |
+		Promise<ApolloQueryResult<MotionPostAndCommentsQuery>> |
+		Promise<ApolloQueryResult<ReferendumPostAndCommentsQuery>> |
+		Promise<ApolloQueryResult<ProposalPostAndCommentsQuery>> |
+		Promise<ApolloQueryResult<DiscussionPostAndCommentsQuery>>
+}
+
+const ReactionButton = function ({
+	className,
+	reactionId,
+	reaction,
+	count,
+	userMap,
+	postId,
+	commentId,
+	refetch
+}: ReactionButtonProps) {
+	const { id } = useContext(UserDetailsContext);
+	const [addPostReactionMutation] = useAddPostReactionMutation();
+	const [addCommentReactionMutation] = useAddCommentReactionMutation();
+	const [deletePostReactionMutation] = useDeletePostReactionMutation();
+	const [deleteCommentReactionMutation] = useDeleteCommentReactionMutation();
+	const { queueNotification } = useContext(NotificationContext);
+	const reacted = !!userMap[`${id}`];
+
+	const handleReact = () => {
+		if (!id) {
+			queueNotification({
```

Comment:
Rather than letting users think they can react, and then fail, I'd disable the buttons if not logged-in, and here just fail with a console.error just in case.

Reply:
how to show the disabled button?

Reply:
I guess buttons have a "disable" props [here](https://github.com/paritytech/polkassembly/pull/555/files#diff-22e46fc915c1220f47aabd9b2534dc3cR132). We can figure out the styling.

Reply:
done